cfme/tests/intelligence/reports/test_views.py

```diff
@@ -46,6 +44,10 @@ def test_report_view(report, view_mode):
         caseimportance: high
         initialEstimate: 1/6h
     """
-    view = navigate_to(report, 'Details')
+    try:
+        view = navigate_to(report, 'Details')
+    except CandidateNotFound:
+        report.browser.refresh()
```

Comment:
Q: can this be achieved using `wait_for`?

Reply:
I receive CandidateNotFound error while navigating to that page, I am not sure how to use `wait_for` in this case, can you show me an example of it?

Reply:
Looks like it didn't work @digitronik. Do you have any other way on your mind?

Reply:
@valaparthvi I'm curious, if you navigate manually to this page, are all elements present at the first time?

Reply:
@lina-nikiforova There are no problems with other elements of the page, so they are present the first time, but here's the thing: If you check this [video](https://up1-vrutkovs.int.open.paas.redhat.com/#HBTUFO0ssN7NIDLt7Cmedw), when the report is queued, the report name shown in tree in `Generating report`, the page is being continuously refreshed until a proper name of report is generated, this name appears to be updated in the saved_reports but sometimes it does not reflect in the tree, which is why we encounter CandidateNotFound error, for which we need to refresh the page.

Reply:
@digitronik I have added exception handling inside the navigation step and you might see some piece of code getting repeated, but I am working on making it DRY.

Reply:
@valaparthvi I think this is not way... I understand you need browser refresh.
last try which is in my mind... can you please try this.
`prerequisite` is `Details` page of `report`. add resetter for detail like
```
    def resetter(self, *args, **kwargs):
        self.view.browser.refresh()
```